### ISSUE TITLE
LM getting stuck between two poses

### ISSUE THREAD

**Issue by Dysl3xik** [SMRT-AIST/fast_gicp#70]
I have had issues with some point data where LM optimization gets stuck jumping between two poses where the delta between them is sufficient enough that it does not terminate the search, and instead runs until num_iterations and terminates "without converging"

I added a check to cache every other pose then compare that to the current pose and if the delta is still consider the algorithm to be converged. I am unsure how/why the optimization is jumping between these two points, but its something that seems to happen with my data sets pretty regularly.

I can send you a PR of this hacky fix unless you have an idea what might be happening with the LM. Its also worth noting this only happens when I have a very small transformation_epsilon_ 1e-9. But this number is not atypical for other registration limits and the algorithm usually works with that limit in most cases.

**Comment by koide3** [comment]
I guess ```transformation_epsilon=1e-9``` was too small, and LM failed to decrease the objective function due to numerical errors with that setting. Can you see the LM optimization progress by setting ```gicp.setDebugPrint(true)```?

**Comment by Dysl3xik** [comment]
Yes I have printed it out with, it generally gets stuck in a loop around 1e-5. However if I use that number I generally slightly worse results.

**Comment by koide3** [comment]
I have no idea what is the cause of the problem. Is it possible for you to share some example data to reproduce it?

> I can send you a PR of this hacky fix unless you have an idea what might be happening with the LM.

I would be happy if you could open a PR to fix it.
